keps/sig-cli/0031-kustomize-integration.md

```diff
@@ -167,30 +256,151 @@ To get the detailed objects in a kustomization
+- There is a risk that publishing each command as a separately built binary could cause the aggregate download
+  size of the toolset to balloon.  The kubectl binary is *52M* and the kustomize binary is *31M*.  (extrapolate to
+  30+ commands x 30MB).  Before going down this route, we should consider how to we might want to design a solution
+  and the tradeoffs.    
+
+**AI's for kubectl extensibility:**
+
+- Clearly define the tradeoffs of designing kubectl around plugins - what are the benefits and what are the risks.
+- Evaluate approaches for distributing designing kubectl around a plugin architecture.
+- Evaluate approaches for building, testing, upgrading kubectl around a plugin architecture.
+- If feasible - prioritize and possibly come up with a roadmap for supporting these options towards this approach.
+- Evaluate good, low risk candidate commands.
+
+## Kustomize Example
+
+Following is an example of a kustomization.yaml file used by Kustomize:
 
-To enable kustomize in kubectl, the function `FilenameParam` inside Builder type will be updated to recognize kustomization directories. The Builder will expand the sources in a kustomization directory and pass them to a subcommand.
+```
+apiVersion: v1beta1
+kind: Kustomization
+namePrefix: alices-
 
- Since kustomization directories themselves have a recursive structure, `-R` will be ignored on those directories. Allowing recursive visit to the same files will lead to duplicate resources. 
+commonAnnotations:
+  oncallPager: 800-555-1212
 
-The examples and descriptions for apply, get and delete will be updated to include the support of kustomization directories. 
+configMapGenerator:
+- name: myJavaServerEnvVars
+literals:  
+- JAVA_HOME=/opt/java/jdk
+- JAVA_TOOL_OPTIONS=-agentlib:hprof
+
+secretGenerator:
+- name: app-sec
+commands:
+  username: "echo admin"
+  password: "echo secret"
+```
+
+The result of running `kustomize build` on this sample kustomizaiton.yaml file is:
+
+```
+apiVersion: v1
+data:
+JAVA_HOME: /opt/java/jdk
+JAVA_TOOL_OPTIONS: -agentlib:hprof
+kind: ConfigMap
+metadata:
+annotations:
+  oncallPager: 800-555-1212
+name: alices-myJavaServerEnvVars-7bc9c27cmf
+---
+apiVersion: v1
+data:
+password: c2VjcmV0Cg==
+username: YWRtaW4K
+kind: Secret
+metadata:
+annotations:
+  oncallPager: 800-555-1212
+name: alices-app-sec-c7c5tbh526
+type: Opaque
+```
+
+### Implementation Details/Notes/Constraints
+
+In contrast to `kubectl apply`, which was developed directly in kubernetes/kubernetes and had minimal usage or
+real world experience prior, Kustomize was built as a outside of kubernetes/kubernetes in the
+kubernetes-sigs/kustomize repo.  After `kubectl apply` was published, many issues were uncovered in it that should
+have been discovered earlier.  Implementing Kustomize independently allowed more time for gathering feedback and
+identifying issues.
+
+Kustomize library code will be moved from its current repository (location) to the cli-runtime repository used by
```

Comment:
Wait, isn't that dependent on the actual decision. The KEP oulines 3 possible options, but doesn't pick a winner, but it should so that we can clearly state further actions like this one.

Reply:
Updated